### ISSUE TITLE
Fail to replace map property on document

### ISSUE THREAD

**Issue by kaptnkoala** [cbl-dart/cbl-dart#220]
**Describe the bug**
Why is it intended, that values can not be applied if the old value is a CollectionFLValue? 

I try to replace a nested map on a document with a changed version of the map, but the check in `valueWouldChange` for `CollectionFLValue` always return sfalse, so the new map is never applied.

And what is the best approach, so replace multiple parts of the document, when i have already identified which parts have changed upfront? Thinking about a method like `replaceData` or `merge` that skips the clearing part of `setData`.

**To Reproduce**
1. Create a document with a map/dictionary on one property
2. Replace the current map with a new version using `MutableDocument.setValue`

**Environment (please complete the following information):**
 - OS: iOS / Android
 - Version: 
 - `cbl`: 1.0.0-beta.7
 - `cbl_flutter`: 1.0.0-beta.7

**Comment by blaugold** [comment]
That's definitely a bug. This branch of `valueWouldChange` should always return `true`.

The most efficient way to update a document is to make the changes as small as possible, so setting individual properties would be optimal. The next best thing is to replace the parts of a document that have changes in them.

```dart
final doc = MutableDocument({
  'age': 22,
  'eyeColor': 'brown',
  'name': {
    'first': 'Alice',
    'last': 'Barnes',
  },
});

doc['name'].value = {
  'first': 'Bob',
  'last': 'Barnes',
};
```

In this [example](https://github.com/cbl-dart/cbl-dart/blob/46da1b0a9ad6dd887afc77c960f41e1e2e162e1a/packages/cbl_flutter/example/lib/main.dart#L211) I have sketched out how you could create models that are backed by a document. The example only demonstrates an immutable view of a model, but it could be extended to mutable models. The implementation of these kinds of models could also be generated, similar to how JSON de/serialization code is generated by other packages.

**Comment by blaugold** [comment]
The fix has been published in `cbl: 1.0.0-beta.8`.
